### DISCUSSION TITLE
make launch failure windows

### DISCUSSION THREAD

**Discussion by 11rahi23** [carla-simulator/carla#5763] [Q&A]
Hi,

I am trying to build Carla following the documentations (https://carla.readthedocs.io/en/latest/build_windows/). After following all steps, when I try "make launch", the following errors appear. 

 D:\Carla\carla\Unreal\CarlaUE4\Plugins\Carla\Source\Carla\Game\TaggedComponent.h(83): error C2504: 'FHierarchicalStaticMeshSceneProxy': base class undefined
  D:\Carla\carla\Unreal\CarlaUE4\Plugins\Carla\Source\Carla\Game\TaggedComponent.h(87): error C3668: 'FTaggedHierarchicalStaticMeshSceneProxy::GetViewRelevance': method with override specifier 'override' did not override any base class methods
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Game/TaggedComponent.cpp(170): error C2440: 'return': cannot convert from 'FTaggedHierarchicalStaticMeshSceneProxy *' to 'FPrimitiveSceneProxy *'
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Game/TaggedComponent.cpp(170): note: Types pointed to are unrelated; conversion requires reinterpret_cast, C-style cast or function-style cast
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Game/TaggedComponent.cpp(237): error C2039: 'NotDrawTaggedComponents': is not a member of 'FEngineShowFlags'
  C:\Program Files\Epic Games\UE_4.26\Engine\Source\Runtime\Engine\Classes\Components/BillboardComponent.h(13): note: see declaration of 'FEngineShowFlags'
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Game/TaggedComponent.cpp(265): error C2039: 'NotDrawTaggedComponents': is not a member of 'FEngineShowFlags'
  C:\Program Files\Epic Games\UE_4.26\Engine\Source\Runtime\Engine\Classes\Components/BillboardComponent.h(13): note: see declaration of 'FEngineShowFlags'
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Game/TaggedComponent.cpp(291): error C2039: 'NotDrawTaggedComponents': is not a member of 'FEngineShowFlags'
  C:\Program Files\Epic Games\UE_4.26\Engine\Source\Runtime\Engine\Classes\Components/BillboardComponent.h(13): note: see declaration of 'FEngineShowFlags'
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Game/TaggedComponent.cpp(300): error C2614: 'FTaggedHierarchicalStaticMeshSceneProxy': illegal member initialization: 'FHierarchicalStaticMeshSceneProxy' is not a base or member
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Game/TaggedComponent.cpp(305): error C2065: 'bVerifyUsedMaterials': undeclared identifier
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Game/TaggedComponent.cpp(307): error C2065: 'FLODInfo': undeclared identifier
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Game/TaggedComponent.cpp(307): error C2065: 'LODInfo': undeclared identifier
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Game/TaggedComponent.cpp(307): error C2143: syntax error: missing ';' before ':'
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Game/TaggedComponent.cpp(307): error C2065: 'LODs': undeclared identifier
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Game/TaggedComponent.cpp(307): error C2143: syntax error: missing ';' before ')'
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Game/TaggedComponent.cpp(308): error C2653: 'FLODInfo': is not a class or namespace name
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Game/TaggedComponent.cpp(308): error C2530: 'SectionInfo': references must be initialized
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Game/TaggedComponent.cpp(308): error C2143: syntax error: missing ';' before ':'
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Game/TaggedComponent.cpp(308): error C2065: 'LODInfo': undeclared identifier
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Game/TaggedComponent.cpp(308): error C2143: syntax error: missing ';' before ')'
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Game/TaggedComponent.cpp(316): error C2653: 'FHierarchicalStaticMeshSceneProxy': is not a class or namespace name
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Game/TaggedComponent.cpp(318): error C2039: 'NotDrawTaggedComponents': is not a member of 'FEngineShowFlags'
  C:\Program Files\Epic Games\UE_4.26\Engine\Source\Runtime\Engine\Classes\Components/BillboardComponent.h(13): note: see declaration of 'FEngineShowFlags'
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Recorder/CarlaRecorder.cpp(476): warning C4458: declaration of 'Role' hides class member
  C:\Program Files\Epic Games\UE_4.26\Engine\Source\Runtime\Engine\Classes\GameFramework/Actor.h(541): note: see declaration of 'AActor::Role'
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Recorder/CarlaRecorder.cpp(491): warning C4458: declaration of 'Role' hides class member
  C:\Program Files\Epic Games\UE_4.26\Engine\Source\Runtime\Engine\Classes\GameFramework/Actor.h(541): note: see declaration of 'AActor::Role'
  [4/8] Module.Carla.2_of_3.cpp
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Recorder/CarlaRecorderQuery.cpp(51): warning C4996: 'localtime': This function or variable may be unsafe. Consider using localtime_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details.
  D:\Carla\carla\Unreal\CarlaUE4\Plugins\Carla\Source\Carla/Vehicle/CarlaWheeledVehicle.h(61): warning C4996: 'AWheeledVehicle': PhysX is deprecated. Use the AWheeledVehiclePawn from the ChaosVehiclePhysics Plugin. Please update your code to the new API before upgrading to the next release, otherwise your project will no longer compile.
  D:\Carla\carla\Unreal\CarlaUE4\Plugins\Carla\Source\Carla/Vehicle/CarlaWheeledVehicle.h(161): warning C4996: 'UWheeledVehicleMovementComponent4W': PhysX is deprecated. Use the UChaosWheeledVehicleMovementComponent from the ChaosVehiclePhysics Plugin. Please update your code to the new API before upgrading to the next release, otherwise your project will no longer compile.
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Sensor/CollisionSensor.cpp(49): warning C4458: declaration of 'Episode' hides class member
  D:\Carla\carla\Unreal\CarlaUE4\Plugins\Carla\Source\Carla/Sensor/Sensor.h(115): note: see declaration of 'ASensor::Episode'
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Sensor/DVSCamera.cpp(158): warning C4458: declaration of 'Stream' hides class member
  D:\Carla\carla\Unreal\CarlaUE4\Plugins\Carla\Source\Carla/Sensor/Sensor.h(111): note: see declaration of 'ASensor::Stream'
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Sensor/GnssSensor.cpp(60): warning C4458: declaration of 'Stream' hides class member
  D:\Carla\carla\Unreal\CarlaUE4\Plugins\Carla\Source\Carla/Sensor/Sensor.h(111): note: see declaration of 'ASensor::Stream'
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Sensor/InertialMeasurementUnit.cpp(48): warning C4458: declaration of 'Owner' hides class member
  C:\Program Files\Epic Games\UE_4.26\Engine\Source\Runtime\Engine\Classes\GameFramework/Actor.h(523): note: see declaration of 'AActor::Owner'
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Sensor/InertialMeasurementUnit.cpp(187): warning C4458: declaration of 'Stream' hides class member
  D:\Carla\carla\Unreal\CarlaUE4\Plugins\Carla\Source\Carla/Sensor/Sensor.h(111): note: see declaration of 'ASensor::Stream'
  D:\Carla\carla\Unreal\CarlaUE4\Plugins\Carla\Source\Carla/Game/TaggedComponent.h(83): error C2504: 'FHierarchicalStaticMeshSceneProxy': base class undefined
  D:\Carla\carla\Unreal\CarlaUE4\Plugins\Carla\Source\Carla/Game/TaggedComponent.h(87): error C3668: 'FTaggedHierarchicalStaticMeshSceneProxy::GetViewRelevance': method with override specifier 'override' did not override any base class methods
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Sensor/InstanceSegmentationCamera.cpp(35): error C2039: 'SetNotDrawTaggedComponents': is not a member of 'FEngineShowFlags'
  C:\Program Files\Epic Games\UE_4.26\Engine\Source\Runtime\Engine\Classes\Components/StaticMeshComponent.h(30): note: see declaration of 'FEngineShowFlags'
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Sensor/ObstacleDetectionSensor.cpp(137): warning C4458: declaration of 'Episode' hides class member
  D:\Carla\carla\Unreal\CarlaUE4\Plugins\Carla\Source\Carla/Sensor/Sensor.h(115): note: see declaration of 'ASensor::Episode'
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Sensor/Radar.cpp(136): error C2039: 'ParallelLineTraceSingleByChannel': is not a member of 'UWorld'
  C:\Program Files\Epic Games\UE_4.26\Engine\Intermediate\Build\Win64\UE4Editor\Inc\Engine\GameplayStatics.generated.h(51): note: see declaration of 'UWorld'
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Sensor/Radar.cpp(136): error C2039: '__this': is not a member of 'UWorld'
  C:\Program Files\Epic Games\UE_4.26\Engine\Intermediate\Build\Win64\UE4Editor\Inc\Engine\GameplayStatics.generated.h(51): note: see declaration of 'UWorld'
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Sensor/Radar.cpp(136): error C2737: 'Hitted': const object must be initialized
  D:/Carla/carla/Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Sensor/RayCastSemanticLidar.cpp(229): error C2039: 'ParallelLineTraceSingleByChannel': is not a member of 'UWorld'
  C:\Program Files\Epic Games\UE_4.26\Engine\Intermediate\Build\Win64\UE4Editor\Inc\Engine\GameplayStatics.generated.h(51): note: see declaration of 'UWorld'
make: *** [CarlaUE4Editor] Error 6

Carla version: 0.9.13
UE Version: 4.26.2
Windows 10 

I am stuck with this issue for two days. Will be really grateful if someone helps in solving the problem.

Thanks

**Comment by ishaan95** [comment]
Did you install UE4 according to the CARLA docs [instructions](https://carla.readthedocs.io/en/latest/build_windows/#unreal-engine)? They have their own version of Unreal Engine which is modified.

**Reply by 11rahi23** [reply]
Yes, I installed UE4 according to the CARLA docs instructions, and also changed environment variables accordingly.

**Reply by ishaan95** [reply]
Most of errors are "not a member of" or "base class undefined" which would suggest there is some missing Unreal Engine side code.  The UE4 Carla uses from 0.9.12 onwards comes from here - [https://github.com/CarlaUnreal/UnrealEngine](https://github.com/CarlaUnreal/UnrealEngine)

**Comment by MoolixLin** [comment]
Have you solved this problem?  I meet the same problem as you.

**Reply by 11rahi23** [reply]
I could not install it on windows. Later I installed Ubuntu 16.04 and built Carla there.
